### DISCUSSION TITLE
Can this Sensor Driver be Contrubition to Zephyr

### DISCUSSION THREAD

**Discussion by zafersn** [zephyrproject-rtos/zephyr#65197] [General] [answered]
Hi.

I worked on the LPS22DF sensor driver and wanted to contribute it to Zephyr. But I am not sure if it is worth or suitable for the zephyr because there is a similar driver (lps22hh) in the zephyr sensor driver but it is not compatible(registers etc...) with the lps22df, So people need to edit it to use it. 

and I also noticed during the development of this driver that there is a missing configuration in the `C:\netfeasaproject\zephyr\modules\Kconfig.st` people need to edit this one too. I mentioned about this in this issue https://github.com/zephyrproject-rtos/zephyr/discussions/65096

Here is the register address comparison for lps22df and lps22hh.
LPS22DF:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/13236364/fc1f3949-4ff0-43cf-8a0d-f37328ea6993)

LPS22HH:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/13236364/67259264-e75e-4ff5-b482-c5eeb28d192d)

Moreover, here ctrl register 1 comparison: they are different too.

LPS22HH:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/13236364/ddd41141-5af1-463e-bd59-37bacfa9f0c1)

LPS22DF:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/13236364/2d4236fa-8e67-44a0-ba1b-58da54bce897)

So wondering if this contribution is suitable/worth it for the zephyr?

If so,  these are the steps to contribute: https://docs.zephyrproject.org/latest/contribute/index.html , right?

But I wonder, if there is a short way to contribute, I basically edited/modified an existing driver(lps22hh), changes are: invoking the relevant functions from the st lib which already exist in the module (C:\zephyrproject\modules\hal\st\sensor\stmemsc\lps22df_STdC\driver\lps22df_reg.h) and configurations.

I intended to have a Zephyr Contributor Badge. So if this one is acceptable, it will be the first step for me.

Cheers!

**Comment by kartben** [comment, marked as answer]
As a matter of fact, I believe support for this sensor was added just today! https://github.com/zephyrproject-rtos/zephyr/pull/62028
But to answer your question, yes, of course, we would (have) welcome such a contribution :)
